### PR TITLE
linux-rt_6_1: init at 6.1.19-rt8

### PR DESCRIPTION
**Pull request by vifino**
Also sets the linux_rt_latest to 6.1.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by vifino** [comment]
Hey @orivej, as promised - 6.1 set as latest.

Should I add a commit to set 5.15 as default?

File: pkgs/top-level/linux-kernels.nix [hunk 1]
```diff
@@ -169,6 +169,14 @@ in {
       ];
     };
 
+    linux_rt_6_1 = callPackage ../os-specific/linux/kernel/linux-rt-6.1.nix {
+      kernelPatches = [
+        kernelPatches.bridge_stp_helper
+        kernelPatches.request_key_helper
+        kernelPatches.export-rt-sched-migrate
```
Comment (orivej) [inline pkgs/top-level/linux-kernels.nix:176]: Please add kernelPatches.fix-em-ice-bonding as in linux_6_1

**Review by orivej — CHANGES_REQUESTED** [review]

**Comment by orivej** [comment]
> Should I add a commit to set 5.15 as default?

No, not in this PR

File: pkgs/top-level/linux-kernels.nix (hunk 1 above)
Comment (vifino) [reply pkgs/top-level/linux-kernels.nix:176]: I thought that was intentional - none of the RT kernels have that patch.
I'll add it.

**Comment by orivej** [comment]
I'm running with this kernel now. It has some serious issues, but when it boots OK it seems to work well.

The issues I encountered:

1. Occasionally it does not boot at all, the screen is stuck at EFI stub messages. I think this is a variation of the next issue. 
2. When it boots, it prints at least one BUG into dmesg (and then the kernel continues to print normal dmesg messages into the console). For example:
```
[    0.347083] BUG: scheduling while atomic: swapper/2/0/0x00010002
[    0.347086] Modules linked in:
[    0.347087] Preemption disabled at:
[    0.347088] [<ffffffff93acd515>] migrate_enable+0x45/0x160
[    0.347106] CPU: 2 PID: 0 Comm: swapper/2 Not tainted 6.1.19-rt8 #1-NixOS
[    0.347108] Hardware name: LENOVO 20FRS11T00/20FRS11T00, BIOS N1FET74W (1.48 ) 11/12/2020
[    0.347109] Call Trace:
[    0.347111]  <IRQ>
[    0.347113]  dump_stack_lvl+0x44/0x5c
[    0.347129]  ? migrate_enable+0x45/0x160
[    0.347131]  __schedule_bug.cold+0x80/0x8d
[    0.347134]  __schedule+0xeb4/0x12c0
[    0.347137]  ? update_load_avg+0x80/0x830
[    0.347140]  schedule_rtlock+0x1e/0x40
[    0.347143]  rtlock_slowlock_locked+0x3d1/0xfa0
[    0.347146]  ? enqueue_task_fair+0x87/0x440
[    0.347150]  rt_spin_lock+0x3f/0x60
[    0.347153]  notify_hwp_interrupt+0x7d/0x110
[    0.347156]  intel_thermal_interrupt+0x13/0x2d0
[    0.347158]  __sysvec_thermal+0x23/0x150
[    0.347175]  sysvec_thermal+0x99/0xc0
[    0.347178]  </IRQ>
[    0.347179]  <TASK>
[    0.347180]  asm_sysvec_thermal+0x16/0x20
[    0.347182] RIP: 0010:cpuidle_enter_state+0xde/0x4b0
[    0.347185] Code: 00 00 31 ff e8 23 9b a2 ff 45 84 ff 74 16 9c 58 0f 1f 40 00 f6 c4 02 0f 85 bb 03 00 00 31 ff e8 a8 91 a9 ff fb 0f 1f 44 00 00 <45> 85 f6 0f 88 e5 01 00 00 49 63 d6 48 8d 04 52 48 8d 04 82 49 8d
[    0.347187] RSP: 0000:ffffa7f680113e90 EFLAGS: 00000246
[    0.347190] RAX: ffff8d40f1700000 RBX: ffffc7f67fd00000 RCX: 0000000000000030
[    0.347191] RDX: 0000000000000002 RSI: ffffffff94a6da01 RDI: ffffffff94a79d73
[    0.347193] RBP: 0000000000000004 R08: fffffffffff0bdd3 R09: 000000002db7298a
[    0.347194] R10: 0000000000000018 R11: 00000000000001d6 R12: ffffffff951bf6c0
[    0.347195] R13: 0000000014b00af0 R14: 0000000000000004 R15: 0000000000000000
[    0.347198]  cpuidle_enter+0x29/0x40
[    0.347199]  do_idle+0x236/0x2c0
[    0.347202]  cpu_startup_entry+0x19/0x20
[    0.347204]  start_secondary+0x117/0x140
[    0.347207]  secondary_startup_64_no_verify+0xe5/0xeb
[    0.347211]  </TASK>
[    0.347214] ------------[ cut here ]------------
```
3. Rarely it fails to initialize WiFi:
```
[   16.701096] iwlwifi 0000:04:00.0: Detected Intel(R) Dual Band Wireless AC 8260, REV=0x201
[   17.742292] iwlwifi 0000:04:00.0: SecBoot CPU1 Status: 0x3040001, CPU2 Status: 0x0
[   17.743035] iwlwifi 0000:04:00.0: WFPM_ARC1_PD_NOTIFICATION: 0xf
[   17.743167] iwlwifi 0000:04:00.0: HPM_SECONDARY_DEVICE_STATE: 0x742
[   17.743182] iwlwifi 0000:04:00.0: WFPM_MAC_OTP_CFG7_ADDR: 0x0
[   17.743196] iwlwifi 0000:04:00.0: WFPM_MAC_OTP_CFG7_DATA: 0x0
[   17.743198] iwlwifi 0000:04:00.0: Collecting data: trigger 15 fired.
[   17.993199] iwlwifi 0000:04:00.0: Not valid error log pointer 0x00000000 for Init uCode
[   17.993326] iwlwifi 0000:04:00.0: IML/ROM dump:
[   17.993327] iwlwifi 0000:04:00.0: 0x00000000 | IML/ROM error/state
[   17.993341] iwlwifi 0000:04:00.0: 0x03040001 | IML/ROM data1
[   17.993349] iwlwifi 0000:04:00.0: Fseq Registers:
[   17.993353] iwlwifi 0000:04:00.0: 0x646BB649 | FSEQ_ERROR_CODE
[   17.993358] iwlwifi 0000:04:00.0: 0x78EF5380 | FSEQ_TOP_INIT_VERSION
[   17.993362] iwlwifi 0000:04:00.0: 0x006A279B | FSEQ_CNVIO_INIT_VERSION
[   17.993367] iwlwifi 0000:04:00.0: 0x0000A056 | FSEQ_OTP_VERSION
[   17.993371] iwlwifi 0000:04:00.0: 0xCCA159D4 | FSEQ_TOP_CONTENT_VERSION
[   17.993376] iwlwifi 0000:04:00.0: 0x789BA961 | FSEQ_ALIVE_TOKEN
[   17.993380] iwlwifi 0000:04:00.0: 0xCD841489 | FSEQ_CNVI_ID
[   17.993384] iwlwifi 0000:04:00.0: 0xE15639B2 | FSEQ_CNVR_ID
[   17.993389] iwlwifi 0000:04:00.0: 0x03000000 | CNVI_AUX_MISC_CHIP
[   17.993396] iwlwifi 0000:04:00.0: 0x0BADCAFE | CNVR_AUX_MISC_CHIP
[   17.993403] iwlwifi 0000:04:00.0: 0x0BADCAFE | CNVR_SCU_SD_REGS_SD_REG_DIG_DCDC_VTRIM
[   17.993411] iwlwifi 0000:04:00.0: 0x0BADCAFE | CNVR_SCU_SD_REGS_SD_REG_ACTIVE_VDIG_MIRROR
[   17.993414] iwlwifi 0000:04:00.0: Failed to start INIT ucode: -110
[   17.993417] iwlwifi 0000:04:00.0: Collecting data: trigger 16 fired.
```

Overall I'm fine with merging this PR as is, I don't expect rt-latest to be a safe choice. I'll try using 5.15-rt for a while to see if it is a good candidate for the default rt kernel, since 6.1 clearly is not ready yet.

**Review by orivej — APPROVED** [review]

**Comment by orivej** [comment]
@vifino, does this kernel work well for you? Do you see any BUG in `dmesg` output?

**Comment by orivej** [comment]
(Currently I'm running on 5.15.96-rt61 without any issues.)

**Comment by orivej** [comment]
Merged, but a usage report would still be useful

**Comment by vifino** [comment]
Hey! Sorry for the late response.

No, while testing on my x86_64 Lenovo ThinkPad P1 Gen2, I did not see any BUG messages, everything works as expected.
I did not get to test my main usecase for it though, which is userspace IP routing with DPDK, but my test setup is currently disfunctional for that.

**Comment by orivej** [comment]
Thanks, good to know!

**Pull request merged**